### PR TITLE
Add webworkerhotupdatechunktemplateplugin tests

### PR DESCRIPTION
**Pull request by alistairjcbrown**
**What kind of change does this PR introduce?**

Add tests for webworkerhotupdatechunktemplateplugin

**Did you add tests for your changes?**

This PR is only tests

**If relevant, link to documentation update:**

N/A

**Summary**

Based on the coveralls report, the `webworkerhotupdatechunktemplateplugin` file has 41% test coverage. There is no `webworkerhotupdatechunktemplateplugin` specific test file - this is added in this PR and aims to achieve 100% test coverage.
https://coveralls.io/builds/9491731/source?filename=lib%2Fwebworker%2FWebWorkerHotUpdateChunkTemplatePlugin.js

**Does this PR introduce a breaking change?**

No

**Other information**

This PR is branched off PR #3684, which introduces the `applyPluginWithOptions` helper function. This helper has been used for testing this plugin too.
This PR also introduces [`sinon`](https://www.npmjs.com/package/sinon) (as [recommended by mocha](https://github.com/mochajs/mocha/wiki/Spies)) to help with checking if callbacks are called, and if they are called with the correct parameters.

### CODE REVIEW

**Comment by sokra** [comment]
Could you rebase it so the merged commit is gone?

**Comment by alistairjcbrown** [comment]
@sokra @TheLarkInn Thanks - branch rebased 👍

**Comment by TheLarkInn** [comment]
Awesome. Thank you.

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "memory-fs": "~0.4.1",
     "mkdirp": "~0.5.0",
     "node-libs-browser": "^2.0.0",
+    "sinon": "^1.17.7",
```
Comment (wtgtybhertgeghgtwtg) [inline package.json]: Why is this a `dependency` instead of a `devDependency`?

Comment (alistairjcbrown) [inline package.json]: @wtgtybhertgeghgtwtg Good spot - it should be in `devDependencies`. I've put up a PR to correct; #3707

**Pull request merged**